design-book/src/design-constraints.md

```diff
@@ -0,0 +1,58 @@
+# Design Constraints
+
+To help ensure that we're building an editor that's both useful to end users and feasible to maintain,
+we've developed a set of design constraints and goals. Please try to follow them when making and reviewing changes!
+As we're developing, initial prototypes may bend or break the rules: just be mindful of how we can move towards the final desirable end goal.
+
+Below each bullet point, motivation is given.
+
+## Must
+
+- The editor must target the latest stable Bevy release.
+  - This eases project management by separating the work across two repositories, and ensures that we aggressively test each Bevy release's migration path.
```

Comment:
I disagree with this.
Released version of the editor should target the latest stable Bevy release, but in development editor should target in development Bevy. And I would prefer to have everything in the same repository

While prototyping, targeting a stable version of Bevy will be easier and keeping the editor in its own repository makes sense. Once we're happy with the first version, it should be brought back in the same repository as Bevy.